src/System.Private.ServiceModel/src/System/ServiceModel/Channels/LayeredChannel.cs

```diff
@@ -57,12 +58,12 @@ protected override void OnClose(TimeSpan timeout)
 
         protected override IAsyncResult OnBeginClose(TimeSpan timeout, AsyncCallback callback, object state)
         {
-            return _innerChannel.BeginClose(timeout, callback, state);
+            throw ExceptionHelper.PlatformNotSupported();
```

Comment:
Good idea, but we experienced some issues with Contract.Requires in other places (i.e. didn't work properly), and we don't do static analysis, so I'm a little reluctant to add it.
Note: based on past experience, if static analysis becomes available and we enable it, I think we can expect the list of contract precondition violations to number in the 10,000's

---
In reply to: [78151294](https://github.com/dotnet/wcf/pull/1515#discussion_r78151294) [](ancestors = 78151294,78079821)